agent/wsclient/client.go

```diff
@@ -127,23 +134,41 @@ func (cs *ClientServerImpl) Connect() error {
 		return err
 	}
 
+	parsedURL.Scheme = wsScheme
+
 	// NewRequest never returns an error if the url parses and we just verified
 	// it did above
-	request, _ := http.NewRequest("GET", cs.URL, nil)
+	request, _ := http.NewRequest("GET", parsedURL.String(), nil)
+
 	// Sign the request; we'll send its headers via the websocket client which includes the signature
-	utils.SignHTTPRequest(request, cs.Region, ServiceName, cs.CredentialProvider, nil)
+	utils.SignHTTPRequest(request, cs.AgentConfig.AWSRegion, ServiceName, cs.CredentialProvider, nil)
 
-	wsConn, err := cs.websocketConn(parsedURL, request)
-	if err != nil {
-		return err
+	timeoutDialer := &net.Dialer{Timeout: wsConnectTimeout}
+	tlsConfig := &tls.Config{ServerName: parsedURL.Host, InsecureSkipVerify: cs.AgentConfig.AcceptInsecureCert}
+
+	// Ensure that NO_PROXY gets set
+	noProxy := os.Getenv("NO_PROXY")
+	dockerHost, err := url.Parse(cs.AgentConfig.DockerEndpoint)
```

Comment:
I think you want to set `dockerHost.Scheme = ""` so that the scheme is omitted when you do `dockerHost.String()`.